Wright_Will/Assignments/login_and _registration/server.py

```diff
@@ -1,35 +1,78 @@
 from flask import Flask, session, redirect, render_template, flash, request
 from mysqlconnection import MySQLConnector
+from flask.ext.bcrypt import Bcrypt
 import re
 app = Flask(__name__)
-app.secret_key = 'fred'
+app.secret_key = 'bob'
 mysql = MySQLConnector(app,'login_reg')
+bcrypt = Bcrypt(app)
 
 @app.route('/')
 def index():
     return render_template('index.html')
 @app.route('/process',methods=['POST'])
 def process():
-    email = request.form['email']
-    fname = request.form['fname']
-    lname = request.form['lname']
-    password = request.form['password']
-    confirm_pass = request.form['confirm_pass']
     error = False
+    formtype = request.form['formtype']
+    password = request.form['password']
+    session['email'] = request.form['email']
+    # register specific session values
+    if formtype == 'register':
+        session['fname'] = request.form['fname']
+        session['lname'] = request.form['lname']
+        confirm_pass = request.form['confirm_pass']
+        # validate first name
+        if not re.match(r"\b[a-zA-Z]{3,}\b",session['fname'],):
+            flash("First Name can only contain letters and must be at least 2 charachters")
+            error = True
+        #validate last name
+        if not re.match(r"\b[a-zA-Z]{3,}\b",session['lname'],):
+            flash("Last Name can only contain letters and must be at least 2 charachters")
+            error = True
 
-    if not re.match(r"\b[a-z]+[a-z0-9_]*@.+\..+\b",email,):
+        if password != confirm_pass:
+            flash("Password and Confirm Password must match")
+            error = True
+    # validate email
+    if not re.match(r"\b[a-z]+[a-z0-9_]*@.+\..+\b",session['email'],):
         flash("not a valid email")
         error = True
-
-    if not re.match(r"\b[a-zA-Z]+\b",fname,):
-        flash("First Name can only contain letters and must be at least 2 charachters")
+    #validate password
+    if not re.match(r"\b[a-zA-Z0-9_]{8,}\b",password,):
+        flash("Password must be alpha-numeric and at least 8 charachters")
         error = True
-
+    user_query = "SELECT * FROM user WHERE email = :email"
+    pw_hash = bcrypt.generate_password_hash(password)
+    query_data = { 'email': session['email'],'fname':session['fname'],'lname':session['lname'] }
```

Comment:
Since your `data` dictionary is identical to your session dictionary, no need to make quey_data. Just submit session to mysql.query_db